test/test_utils.py

```diff
@@ -108,6 +129,9 @@ def assert_expected_wrapper(actual, expected):
         elif isinstance(_actual, tuple):
             # outputs are from multiple layers: (Tensor, Tensor, ...)
             assert_expected_wrapper(tuple_to_dict(_actual), tuple_to_dict(_expected))
+        elif is_named_tuple(_actual):
+            # output is another named tuple instance
+            assert_expected_wrapper(_actual, _expected)
         elif isinstance(_actual, Tensor):
             # single tensor output
             _expected_shape, _expected_sum = _expected
```

Comment:
why are we testing against tensor sums and shapes, is this just to stray away from numerical tests?